network/transports_test.go

```diff
@@ -100,3 +101,41 @@ func TestDialWithBackoff(t *testing.T) {
 	}
 	c.Close()
 }
+
+func TestDialTLSWithBackoff(t *testing.T) {
+	tlsConf := &tls.Config{
+		InsecureSkipVerify: true, // FIXME: Is it possible to set false for the test server?
+		ServerName:         "example.com",
+		MinVersion:         tls.VersionTLS12,
+	}
+	// Nobody's listening on a random port. Usually.
+	c, err := DialTLSWithBackOff(context.Background(), "tcp4", "127.0.0.1:41482", tlsConf)
+	if err == nil {
+		c.Close()
+		t.Error("Unexpected success dialing")
+	}
+
+	// Make the test short.
+	bo := backOffTemplate
+	bo.Steps = 2
+
+	// Timeout. Use special testing IP address.
+	c, err = dialBackOffHelper(context.Background(), "tcp4", "198.18.0.254:8888", bo, tlsConf)
+	if err == nil {
+		c.Close()
+		t.Error("Unexpected success dialing")
+	}
+	const expectedErrPrefix = "timed out dialing"
+	if err == nil || !strings.HasPrefix(err.Error(), expectedErrPrefix) {
+		t.Errorf("Error = %v, want: %s(...)", err, expectedErrPrefix)
+	}
+
+	s := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+	defer s.Close()
+
```

Comment:
Try something like:

```golang

rootCAs := x509.NewCertPool()
rootCAs.AddCert(s.Certificate())
tlsConfig.RootCAs = root as
```

Or, you can initialize `tlsConfig.RootCAs` at the top, and then call `AddCert` here once you have the cert. That would also allow you to add a test of attempting to connect before calling `AddCert` and having the connection fail with a cert mismatch.